sdk/openai/Azure.AI.OpenAI/src/Custom/ChatMessage.Serialization.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Text.Json;
+using Azure;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI
+{
+    [CodeGenSuppress("global::Azure.Core.IUtf8JsonSerializable.Write", typeof(Utf8JsonWriter))]
+    public partial class ChatMessage : IUtf8JsonSerializable
+    {
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("role"u8);
+            writer.WriteStringValue(Role.ToString());
+            // Custom code note: we need to force explicit serialization of 'content' even if it's null.
```

Comment:
Looking at [OpenAI's API reference](https://platform.openai.com/docs/api-reference/chat/create), it seems that this property is required and nullable (although I could have sworn this used to say it was optional? 🤔). Therefore, it should be defined [in the TypeSpec](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp#L37) as:

```
content: string | null;
```

instead of

```
content? : string;
```

Reply:
Good catch. Updated and this did indeed eliminate the need for the custom code here!